src/BcPrebidVast.js

```diff
@@ -0,0 +1,365 @@
+    var node = document.getElementsByTagName('head')[0];
+    node.appendChild(pbjsScr);
+}
+
+// this function loads MailOnline Plugin
+var molLoadingInProgress = false;
+var molLoaded = false;
+function loadMolPlugin(callback) {
+	var vjs = window.videojs || false;
+	if (!vjs) {
+    	_logger.warn(_prefix, 'Videojs is not loaded yet');
+		callback(false);
+		return;
+	}
+	if (!vjs.getPlugins().vastClient) {
+		if (document.getElementById('mol-script')) {
+			if (!molLoadingInProgress) {
+		    	_logger.log(_prefix, 'MailOnline Plugin ' + (molLoaded ? '' : 'not ') + 'loaded successfilly already');
+				callback(molLoaded);
+			}
+			else {
+				var waitMolLoaded = setInterval(function() {
+					if (!molLoadingInProgress) {
+						clearInterval(waitMolLoaded);
+				    	_logger.log(_prefix, 'MailOnline Plugin ' + (molLoaded ? '' : 'not ') + 'loaded successfilly already');
+						callback(molLoaded);
+					}
+				}, 50);
+			}
+			return;
+		}
+		molLoadingInProgress = true;
+	    var molScr = document.createElement('script');
+	    molScr.id = 'mol-script';
+	    molScr.onload = function() {
+	    	_logger.log(_prefix, 'MailOnline Plugin loaded successfilly');
+	    	molLoaded = true;
+	    	molLoadingInProgress = false;
+	    	callback(true);
+	    };
+	    molScr.onerror = function(e) {
+	    	_logger.error(_prefix, 'Failed to load MailOnline Plugin. Error event: ', e);
+	    	molLoadingInProgress = false;
+	    	callback(false);
+	    };
+	    molScr.async = true;
+	    molScr.type = 'text/javascript';
+	    molScr.src = '//acdn.adnxs.com/video/plugins/mol/videojs_5.vast.vpaid.min.js';
+	    var node = document.getElementsByTagName('head')[0];
+	    node.appendChild(molScr);
+	}
+	else {
+    	_logger.log(_prefix, 'MailOnline Plugin already loaded');
+		callback(true);
+	}
+}
+
+(function () {
+	// if bidders settings are present in the $$PREBID_GLOBAL$$.plugin_prebid_options variable load prebid.js and do the bidding
+	if ($$PREBID_GLOBAL$$.plugin_prebid_options && $$PREBID_GLOBAL$$.plugin_prebid_options.biddersSpec) {
+		BC_prebid_in_progress = true;
+		loadPrebidScript($$PREBID_GLOBAL$$.plugin_prebid_options, true);
+	}
+	loadMolPlugin(function() {});
+})();
+
+// register videojs prebid plugins
+function registerPrebidVastPlugin(vjs) {
+	_vjs = vjs;
+	if (!_vjs.getPlugins().bcPrebidVastPlugin) {
+		reqisterPrebidVastPlugin();
+	}
+
+	_vjs.registerPlugin('bcPrebidVastPluginCommand', function(command) {
+		if (command === 'stop') {
+			if (_vastManagerObj) {
+				_vastManagerObj.stop();
+			}
+		}
+	});
+}
+
+var prebidVastPlugin = {
+	// @exclude
+	// Method exposed only for unit Testing Purpose
+	// Gets stripped off in the actual build artifact
+	test: function() {
+		return {
+			doPrebid: function(options, callback) {
+				if ($$PREBID_GLOBAL$$.bc_pbjs === undefined) {
+					loadPrebidScript(options, false);
+					var waitReady = setInterval(function() {
```

Comment:
I think Dave already pointed this out but wait intervals are a bad practice. Callbacks/promises should be preferred

Reply:
This code is not included in final build (located between @exclude and @endexclude macros) and used only for unit test doPrebid function. In this case we don't care about loosing 50 msecs. The code just wait until prebid.js is loaded.